src/topology/sheaves/sheaf_condition/unique_gluing.lean

```diff
@@ -221,22 +220,74 @@ end presheaf
 
 namespace sheaf
 open presheaf
+open category_theory
 
 variables {X : Top.{u}} (F : sheaf (Type u) X) {ι : Type u} (U : ι → opens X)
 
 /--
 A more convenient way of obtaining a unique gluing of sections for a sheaf
 -/
 lemma exists_unique_gluing (sf : Π i : ι, F.presheaf.obj (op (U i)))
- (hsf : is_compatible F.presheaf U sf) :
- ∃! s : F.presheaf.obj (op (supr U)), is_gluing F.presheaf U sf s  :=
+  (h : is_compatible F.presheaf U sf ) :
+  ∃! s : F.presheaf.obj (op (supr U)), is_gluing F.presheaf U sf s :=
 begin
-  have := (sheaf_condition_unique_gluing_of_sheaf_condition _ F.sheaf_condition U sf hsf),
+  have := (sheaf_condition_unique_gluing_of_sheaf_condition _ F.sheaf_condition U sf h),
   refine ⟨this.default.1,this.default.2,_⟩,
   intros s hs,
   exact congr_arg subtype.val (this.uniq ⟨s,hs⟩),
 end
 
+/--
+In this version of the lemma, the inclusion homs `iUV` can be specified directly by the user,
+which can be more convenient in practice.
+-/
+lemma exists_unique_gluing' (V : opens X) (iUV : Π i : ι, U i ⟶ V) (hcover : V ≤ supr U)
+  (sf : Π i : ι, F.presheaf.obj (op (U i))) (h : is_compatible F.presheaf U sf) :
+  ∃! s : F.presheaf.obj (op V), ∀ i : ι, F.presheaf.map (iUV i).op s = sf i :=
+begin
+  have V_eq_supr_U : V = supr U := le_antisymm hcover (supr_le (λ i, le_of_hom (iUV i))),
+  obtain ⟨gl, gl_spec, gl_uniq⟩ := F.exists_unique_gluing U sf h,
+  refine ⟨F.presheaf.map (eq_to_hom V_eq_supr_U).op gl, (λ i,_), (λ gl' gl'_spec,_)⟩,
+  { rw ← functor_to_types.map_comp_apply,
+    exact gl_spec i },
+  { convert congr_arg _ (gl_uniq (F.presheaf.map (eq_to_hom V_eq_supr_U.symm).op gl') (λ i,_)) ;
+      rw ← functor_to_types.map_comp_apply,
+    { exact (functor_to_types.map_id_apply _ _).symm },
+    { convert gl'_spec i }}
+end
+
+@[ext]
+lemma eq_of_locally_eq (s t : F.presheaf.obj (op (supr U)))
+  (h : ∀ i, F.presheaf.map (opens.le_supr U i).op s = F.presheaf.map (opens.le_supr U i).op t) :
+  s = t :=
+begin
+  apply (mono_iff_injective _).mp (mono_of_is_limit_parallel_pair (F.sheaf_condition U)),
+  ext,
+  dsimp [sheaf_condition_equalizer_products.res],
+  simp,
```

Comment:
Could you add the simp lemma that makes this `dsimp` unnecessary?

Reply:
Done: 074e0df6fad6b31e76722bc1502d053e3c7d8fd6